src/models/inbound/dfspConsentRequests.model.ts

```diff
@@ -0,0 +1,281 @@
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Sridhar Voruganti - sridhar.voruganti@modusbox.com
+ --------------
+ ******/
+
+import { PubSub } from '~/shared/pub-sub'
+import { PersistentModel } from '~/models/persistent.model'
+import { StateMachineConfig } from 'javascript-state-machine'
+import { ThirdpartyRequests, Errors } from '@mojaloop/sdk-standard-components';
+import {
+  v1_1 as fspiopAPI,
+  thirdparty as tpAPI
+} from '@mojaloop/api-snippets'
+import inspect from '~/shared/inspect'
+import { HTTPResponseError } from '~/shared/http-response-error'
+import {
+  DFSPConsentRequestsData,
+  DFSPConsentRequestsStateMachine,
+  DFSPConsentRequestsModelConfig
+} from '~/models/inbound/dfspConsentRequests.interface'
+import { DFSPBackendRequests } from '~/shared/dfsp-backend-requests'
+
+// DFSPConsentRequestsModel is the passive inbound handler for inbound
+// POST /consentRequests requests and no response is generated from `model.run()`
+export class DFSPConsentRequestsModel
+  extends PersistentModel<DFSPConsentRequestsStateMachine, DFSPConsentRequestsData> {
+  protected config: DFSPConsentRequestsModelConfig
+
+  constructor (
+    data: DFSPConsentRequestsData,
+    config: DFSPConsentRequestsModelConfig
+  ) {
+    const spec: StateMachineConfig = {
+      init: 'start',
+      transitions: [
+        { name: 'validateRequest', from: 'start', to: 'RequestIsValid' },
+        { name: 'storeReqAndSendOTP', from: 'RequestIsValid', to: 'success' }
+      ],
+      methods: {
+        // specific transitions handlers methods
+        onValidateRequest: () => this.onValidateRequest(),
+        onStoreReqAndSendOTP: () => this.onStoreReqAndSendOTP()
+      }
+    }
+    super(data, config, spec)
+    this.config = { ...config }
+  }
+
+  // getters
+  get pubSub (): PubSub {
+    return this.config.pubSub
+  }
+
+  get dfspBackendRequests (): DFSPBackendRequests {
+    return this.config.dfspBackendRequests
+  }
+
+  get thirdpartyRequests (): ThirdpartyRequests {
+    return this.config.thirdpartyRequests
+  }
+
+  async onValidateRequest (): Promise<void> {
+    const { request, toParticipantId } = this.data
+
+    try {
+      const response = await this.dfspBackendRequests.validateConsentRequests(request)
+
+      if (!response) {
+        throw new Errors.MojaloopFSPIOPError(
+          null as unknown as string,
```

Comment:
for some reason if back-end returns empty response, then will throw exception with specific error code 
we are throwing exception this way in case of validation errors 
Kevin used like that , so followed same pattern.
https://github.com/mojaloop/thirdparty-scheme-adapter/blob/master/src/models/inbound/dfspOTPValidate.model.ts#L108

Reply:
As discussed please write a wrapper for this maybe in utils, or fix upstream.

Reply:
for now I modified that code to `throw new Errors.MojaloopFSPIOPError('', '', '', Errors.MojaloopApiErrorCodes.TP_FSP_CONSENT_SCOPES_ERROR) `
I think it looks fine now,  implementing wrapper or fix upstream require changes in sdk-standard-components.
will do those changes in a separate PR.